http-client/src/test/java/io/airlift/http/client/TestingStatusListener.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.http.client;
+
+import javax.inject.Inject;
+
+import java.util.Map;
+
+public class TestingStatusListener
+        implements HttpStatusListener
+{
+    private final Map<Integer, Integer> statusCounter;
+
+    @Inject
+    public TestingStatusListener(Map<Integer, Integer> statusCounter)
+    {
+        this.statusCounter = statusCounter;
+    }
+
+    @Override
+    public void statusReceived(int statusCode)
+    {
+        statusCounter.compute(statusCode, (ignore, count) -> (count == null) ? 1 : (count + 1));
```

Comment:
With `Multiset` this would simply be `statusCounter.add(statusCode)`. Though you can simplify this for `Map`
```java
statusCounter.merge(statusCode, 1, Math::addExact);